dataland-frontend/tests/component/components/resources/dataTable/conversion/NumberValueGetterFactory.cy.ts

```diff
@@ -1,41 +1,63 @@
+    const value = numberValueGetterFactory("data", sampleNumberFormFieldWithUnit)(dataset);
+    expect(value).to.deep.equal(MLDTDisplayObjectForEmptyString);
   });
 
-  it("The value of the input should be displayed if it exists", () => {
+  it("The value of the input should be displayed together with its unit if it exists", () => {
     const dataset = { data: 10 };
-    const value = numberValueGetterFactory("data")(dataset);
-    expect(value).to.deep.equal(<MLDTDisplayValue<MLDTDisplayComponents.StringDisplayComponent>>{
-      displayComponent: MLDTDisplayComponents.StringDisplayComponent,
-      displayValue: "10",
+    const value = numberValueGetterFactory("data", sampleNumberFormFieldWithUnit)(dataset);
+    expect(value).to.deep.equal(<MLDTDisplayObject<MLDTDisplayComponentName.StringDisplayComponent>>{
+      displayComponentName: MLDTDisplayComponentName.StringDisplayComponent,
+      displayValue: "10 someUnit",
     });
   });
 
-  it("The value of the input should be displayed with a percent sign rounded to two decimal places", () => {
+  it("The value of the input should be displayed rounded to two decimal places", () => {
     const dataset = { data: 10.223 };
-    const value = numberValueGetterFactory("data")(dataset);
-    expect(value).to.deep.equal(<MLDTDisplayValue<MLDTDisplayComponents.StringDisplayComponent>>{
-      displayComponent: MLDTDisplayComponents.StringDisplayComponent,
-      displayValue: "10.22",
+    const value = numberValueGetterFactory("data", sampleNumberFormFieldWithUnit)(dataset);
+    expect(value).to.deep.equal(<MLDTDisplayObject<MLDTDisplayComponentName.StringDisplayComponent>>{
+      displayComponentName: MLDTDisplayComponentName.StringDisplayComponent,
+      displayValue: "10.22 someUnit",
     });
   });
 
   it("The value of the input should contain thousands separators", () => {
     const dataset = { data: 1023 };
-    const value = numberValueGetterFactory("data")(dataset);
-    expect(value).to.deep.equal(<MLDTDisplayValue<MLDTDisplayComponents.StringDisplayComponent>>{
-      displayComponent: MLDTDisplayComponents.StringDisplayComponent,
-      displayValue: "1,023",
+    const value = numberValueGetterFactory("data", sampleNumberFormFieldWithUnit)(dataset);
+    expect(value).to.deep.equal(<MLDTDisplayObject<MLDTDisplayComponentName.StringDisplayComponent>>{
+      displayComponentName: MLDTDisplayComponentName.StringDisplayComponent,
+      displayValue: "1,023 someUnit",
+    });
+  });
+
+  it("The value of the input should be displayed without a unit if it does not have one", () => {
+    const dataset = { data: 1234023.333 };
+    const value = numberValueGetterFactory("data", sampleNumberFormFieldWithoutUnit)(dataset);
+    expect(value).to.deep.equal(<MLDTDisplayObject<MLDTDisplayComponentName.StringDisplayComponent>>{
+      displayComponentName: MLDTDisplayComponentName.StringDisplayComponent,
+      displayValue: "1,234,023.33",
```

Comment:
Is the unit not expected here?

Reply:
No because I use `sampleNumberFormFieldWithoutUnit` which says "WithoutUnit".

My goal is to assure that the number is still shown (without a unit attached to it) if the Field has no unit.